Storage/Storage/SQL/BrowserDB.swift

```diff
@@ -8,11 +8,22 @@ public enum QuerySort {
     case None, LastVisit, Frecency
 }
 
+public class DateRange {
+    let start: NSTimeInterval?
+    let end: NSTimeInterval?
+
+    public init(start: NSTimeInterval?, end: NSTimeInterval?) {
+        self.start = start
+        self.end = end
+    }
+}
+
 public enum FilterType {
     case ExactUrl
     case Url
     case Guid
     case Id
+    case DateRange
```

Comment:
If you add associated types to this enum:

```
enum FilterType {
    case DateRange(DateRange)
}
```

Then you can do something like this:

```
let filter = FilterType.DateRange(DateRange(start: xxx, end: yyy))
```

And where you use the filter:
```
switch filterType {
    case DateRange(let range):
        /// filter on range.start, range.end
}
```

Now you don't need an extra `var filter: AnyObject?` in `QueryOptions` anymore. This is much more type safe then what we do now.